spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/NettyRoutingFilter.java

```diff
@@ -165,4 +165,14 @@ public int getOrder() {
 
 		return responseFlux.then(chain.filter(exchange));
 	}
+
+	private HttpMethod getMethod(ServerHttpRequest request){
+		org.springframework.http.HttpMethod httpMethod = request.getMethod();
+		if(httpMethod != null){
+			return HttpMethod.valueOf(httpMethod.toString());
+		}else {
+			return HttpMethod.valueOf(request.getMethodValue());
```

Comment:
Could not just always do this and forget checking is `getMehtod` returns `null`?

Reply:
I investigated code and saw a lot of checks inside `httpMethod.toString()`. I`m was not shure that all of them are nesessary but left them for safety. So i totally agree that we can forget about checking.

Reply:
I removed unnecessary null checks. Is everything ok now ?